dot/network/notifications.go

```diff
@@ -256,13 +291,38 @@ func (s *Service) broadcastExcluding(info *notificationsProtocol, excluding peer
 				}
 			}
 
+			hsData, has = info.getHandshakeData(peer)
+			if !has {
+				continue
+			}
+
 			// we've already completed the handshake with the peer, send message directly
 			logger.Trace("sending message", "protocol", info.protocolID, "peer", peer, "message", msg)
-			err = s.host.send(peer, info.protocolID, msg)
-		}
 
-		if err != nil {
-			logger.Debug("failed to send message to peer", "peer", peer, "error", err)
+			err = s.host.writeToStream(hsData.stream, msg)
+			if err != nil {
+				logger.Trace("failed to send message to peer", "peer", peer, "error", err)
+			}
 		}
 	}
 }
+
+func readHandshake(stream libp2pnetwork.Stream) (Handshake, error) {
+	msgBytes := make([]byte, maxBlockResponseSize)
+	tot, err := readStream(stream, msgBytes)
+	if err == io.EOF {
+		return nil, io.EOF
+	} else if err != nil {
+		logger.Trace("failed to read handshake from stream", "peer", stream.Conn().RemotePeer(), "protocol", stream.Protocol(), "error", err)
```

Comment:
can remove the logs in this function and leave it to the caller of `readHandshake` to log

Reply:
Done